Docker

Dani-SalanovaUIB opened the pull request:
Para iniciar el proyecto con Docker. Hay que primero crear la imagen:

docker build -t mi-aplicacion .

Y después iniciar el contenedor con dicha imagen:

docker run -d --name mi-contenedor -p 5173:5173 mi-aplicacion

El contenedor actualmente escucha peticiones por el puerto 5173, que es lo que hay en el SecurityConfig.

Dani-SalanovaUIB committed:
inicio de docker

Dani-SalanovaUIB committed:
Puertos cambiados

Indicando los puertos que se usan para comunicarse con la aplicación.

Para crear la imagen:

docker build -t mi-aplicacion .

Para iniciarlo:

docker run -d --name mi-contenedor -p 5173:5173 mi-aplicacion

rmassanet-uib reviewed (changes requested):
Eliminad toda referencia a la BBDD externa, incluso en producción. La BBDD en producción también será local y _dockerizada_.

Dani-SalanovaUIB committed:
Base de datos dockerizada

Ahora hay un fichero docker-compose que genera automaticamente tanto la base de datos como el back.

Para ejecutar:
docker-compose up --build

Para parar:
docker-compsoe down

Gamundi1 committed:
arregla error sincronia

rmassanet-uib reviewed (changes requested):
Cambiad el puerto en el que escucha el back-end al puerto 8081 y así no habrá que cambiar nada en el front-end.

Dani-SalanovaUIB committed:
Cambio de puerto

Ahora el puerto a usar es el 8081

Dani-SalanovaUIB committed:
Merge branch 'docker' of https://github.com/Laboratori-de-Projectes-de-Software/back-end into task/docker

alberba commented on docker-compose.yaml:
```diff
@@ -0,0 +1,37 @@
+    environment:
+      SPRING_DATASOURCE_URL: "jdbc:mysql://db:3306/human_traits?useSSL=false&serverTimezone=UTC&allowPublicKeyRetrieval=true"
+      SPRING_DATASOURCE_USERNAME: "root"
+      SPRING_DATASOURCE_PASSWORD: "putauib777"
```
xd
